learn/models/conformal-regression/index.qmd

```diff
@@ -0,0 +1,437 @@
+
+Conformalized quantile regression ([Romano _et al_ (2019)](https://scholar.google.com/scholar?hl=en&as_sdt=0%2C5&q=%22Conformalized+quantile+regression%22)) is a method to produce intervals that can properly scale the intervals based on what was observed in the training data. It fits an initial [quantile regression](https://en.wikipedia.org/wiki/Quantile_regression) model to do so and also required a split data set, such as our calibration data. 
+
+The original work used basic quantile regression models. For tidymodels, we'll use [quantile regression forests](https://scholar.google.com/scholar?hl=en&as_sdt=0%2C5&q=%22Quantile+Regression+Forests%22). The reason being that some models require nonlinear terms, interactions, and other features to model the data adequately. Users would have to have a parallel modeling process for that model. Random forests for quantile regression are very adaptable, low maintenance, and are robust to tuning parameter specification. There is a minor downside, as seen below. 
+
+The function for this, `int_conformal_quantile()`, has a slightly different interface than the previous functions: 
+
+* It requires both the training and calibration sets.
+* The confidence level must be set in advance. 
+
+We'll also pass an argument to the `quantregForest()` function that will use 2,000 trees to make the forest: 
+
+```{r}
+#| label: quant-pred
+
+quant_int <-
+  int_conformal_quantile(
+    nnet_variable_fit, 
+    train_data = train_variable_data,
+    cal_data = cal_variable_data, 
+    level = 0.90,
+    ntree = 2000)
+quant_int
+
+test_quant_res <- 
+  predict(quant_int, test_variable_data) %>% 
+  bind_cols(test_variable_data)
+
+test_quant_res %>% slice(1:5)
+```
+
+The results: 
+
+```{r}
+#| label: quant-pred-split
+#| echo: false
+#| out-width: 70%
+#| fig-width: 6
+#| fig-height: 4
+
+test_quant_res %>% 
+  ggplot(aes(x)) +
+  geom_point(aes(y = y), alpha = 1 / 20) +
+  geom_line(
+    data = nnet_variable_pred, aes(y = .pred),
+    linewidth = 1, col = "blue"
+  ) +
+  geom_ribbon(aes(ymin = .pred_lower, ymax = .pred_upper), 
+              col = "#D95F02", linewidth = 3 / 4, fill = NA)
+```
+
+The bumpiness of the bound is the consequence of using a tree-based model for the quantiles. Despite that, the intervals do adapt their widths to suit the data. The coverage is also close to the target: `r round(coverage(test_quant_res)$coverage, 1)`%. 
+
+## Coverage
+
+Speaking of coverage, there's a [GitHub repo](https://github.com/topepo/conformal_sim) where we did simulations for these methods to determine their average coverage. The repository's readme file has a summary. 
+
+## Looking Forward
+
+This is a growing research area; we will do our best to implement important new methods as they are created. 
+
+If you are interested and would like to learn more, try these resources: 
```

Comment:
While it's great to have places to learn more, this reminds me of "Never be the keeper of the list".